clang/lib/Lex/PPDirectives.cpp

```diff
@@ -1995,13 +1995,31 @@ void Preprocessor::HandleIncludeDirective(SourceLocation HashLoc,
     auto Include = CActions->handleIncludeDirective(
         *this, IncludePos, CurLexer->getSourceLocation());
 
+    SmallString<128> FilenameBuffer;
+    StringRef Filename = getSpelling(FilenameTok, FilenameBuffer);
+    SourceLocation CharEnd = FilenameTok.getEndLoc();
+    CharSourceRange FilenameRange =
+        CharSourceRange::getCharRange(FilenameTok.getLocation(), CharEnd);
+    bool isAngled =
+        GetIncludeFilenameSpelling(FilenameTok.getLocation(), Filename);
```

Comment:
There's a case for `if (is_style_posix(BackslashStyle) && LangOpts.MicrosoftExt)` where we call `llvm::sys::path::native` on the path we pass into the callback normally, should we also do that here?